runtime/expr/udf.go

```diff
@@ -0,0 +1,41 @@
+package expr
+
+import (
+	"github.com/brimdata/zed"
+	"golang.org/x/exp/slices"
+)
+
+const maxStackDepth = 10_000
+
+type UDF struct {
+	Body Evaluator
+}
+
+func (u *UDF) Call(ectx zed.Allocator, args []zed.Value) *zed.Value {
+	stack := 1
+	if f, ok := ectx.(*frame); ok {
+		stack += f.stack
+	}
+	if stack > maxStackDepth {
+		panic("stack overflow")
```

Comment:
Rather than panic here, do you think we should return a Zed error that includes the name of the current function?